src/App.tsx

```diff
@@ -1,61 +1,124 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+const getMaxTodoId = (todos: Omit<Todo, 'user'>[]) => {
+  const todoIds = todos.map((todo) => todo.id);
+
+  return Math.max(...todoIds);
+};
+
+const populateTodosWithUser = (todos: Omit<Todo, 'user'>[]) => {
+  return todos.map((todo) => {
+    const findUser = usersFromServer.find((user) => user.id === todo.userId);
+
+    return { ...todo, user: findUser || null };
+  });
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isValidTitle, setIsValidTitle] = useState(true);
+  const [isValidUser, setIsValidUser] = useState(true);
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    if (event.target.value) {
+      setIsValidTitle(true);
+    }
+  };
+
+  const handleUserIdChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+
+    if (event.target.value) {
+      setIsValidUser(true);
+    }
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const isValidTitleField = title.length > 0;
+    const isValidUserField = !!userId;
+
+    setIsValidTitle(isValidTitleField);
+    setIsValidUser(isValidUserField);
+
+    if (isValidTitleField && isValidUserField) {
+      setTodos((prevTodos) => {
+        const newTodo = {
+          id: getMaxTodoId(prevTodos) + 1,
+          title,
+          completed: false,
+          userId,
+        };
+
+        return [...prevTodos, newTodo];
+      });
+
+      setTitle('');
+      setUserId(0);
+    }
```

Comment:
handle each error cases - after that make a new todo
```suggestion
    if (!isValidTitleField) {
      setIsValidTitle(false);
    }
    
    if (!isValidUserField) {
      setIsValidUser(false);
    }
    
    if (!isValidTitleField && !isValidUserField) {
      return;
    }

    setTodos((prevTodos) => {
      const newTodo = {
        id: getMaxTodoId(prevTodos) + 1,
        title,
        completed: false,
        userId,
      };

      return [...prevTodos, newTodo];
    });

    setTitle('');
    setUserId(0);
  }
```